src/App.tsx

```diff
@@ -1,223 +1,196 @@
+        case FilterOptions.ACTIVE:
+          return todo.completed === false;
+
+        case FilterOptions.COMPLETED:
+          return todo.completed === true;
+
+        case FilterOptions.ALL:
+        default:
+          return true;
+      }
+    });
+  }, []);
 
-            <span data-cy="TodoTitle" className="todo__title">HTML</span>
-            <button
-              type="button"
-              className="todo__remove"
-              data-cy="TodoDeleteButton"
-            >
-              ×
-            </button>
-
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
+  const visibleTodos = useMemo(() => {
+    return filterBySelect(todos, filtredBy);
+  }, [todos, filtredBy]);
 
-            <span data-cy="TodoTitle" className="todo__title">CSS</span>
+  const amountOfActiveTodos = useMemo(() => {
+    return filterBySelect(todos, FilterOptions.ACTIVE).length;
+  }, [todos]);
 
-            <button
-              type="button"
-              className="todo__remove"
-              data-cy="TodoDeleteButton"
-            >
-              ×
-            </button>
-
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
-
-            <form>
-              <input
-                data-cy="TodoTitleField"
-                type="text"
-                className="todo__title-field"
-                placeholder="Empty todo will be deleted"
-                defaultValue="JS"
-              />
-            </form>
-
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
+  const isClearNeeded = useMemo(() => {
+    return amountOfActiveTodos !== visibleTodos.length;
+  }, [amountOfActiveTodos, visibleTodos]);
```

Comment:
let's simplify our code and get rid of that

Reply:
and you don't need useMemo here

Reply:
I've remove this variable in the third part